core/trie/node_test.go

```diff
@@ -170,5 +171,5 @@ func TestNodeHash(t *testing.T) {
 	}
 	path := bitset.FromWithLength(6, []uint64{42})
 
-	assert.Equal(t, true, expected.Equal(node.Hash(path)), "TestTrieNode_Hash failed")
+	assert.Equal(t, true, expected.Equal(node.Hash(path, crypto.Pedersen)), "TestTrieNode_Hash failed")
```

Comment:
Please add a case for Poseidon

Reply:
That would be testing the Poseidon hash's implementation, not TrieNode.Hash. Testing for each dependency (hash function in this case) defeats the purpose of dependency injection.  

and we don't have a test vector for Poseidon tries.